pkg/storage/replica.go

```diff
@@ -2575,6 +2583,11 @@ func (r *Replica) executeReadOnlyBatch(
 		return nil, roachpb.NewError(err)
 	}
 
+	r.mu.Lock()
+	mergeCompleteCh := r.mu.mergeComplete
+	r.mu.Unlock()
+	<-mergeCompleteCh
```

Comment:
This is the part I'm most iffy about. Is this an ok thing to stick on the command path?